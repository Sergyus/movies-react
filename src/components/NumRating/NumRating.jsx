import React from 'react';
import PropTypes from 'prop-types';
import styles from './NumRating.scss';

const NumRating = (props) => {
  const { rating } = props;
  return (
    <>
      {
        rating
          ? (
            <span className={styles.numRating}>
              {rating}
            </span>
          )
          : null
      }
    </>
  );
};


NumRating.propTypes = {
  rating: PropTypes.number,
};

NumRating.defaultProps = {
  rating: null,
};

export default NumRating;
