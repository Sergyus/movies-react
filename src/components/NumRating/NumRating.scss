@import "../../styles/variables.scss";

.numRating {
  border: 1px solid $blue-color;
  padding: 3px 7px;
  border-radius: 5px;
  font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
  font-weight: bold;
}
