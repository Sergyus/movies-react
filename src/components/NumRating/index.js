import NumRating from './NumRating';

export default NumRating;
