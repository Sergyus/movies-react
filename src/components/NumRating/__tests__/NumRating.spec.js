import React from 'react';
import { create } from 'react-test-renderer';
import NumRating from '..';

describe('NumRating component renders correctly', () => {
  it('NumRating renders correctly', () => {
    const tree = create(<NumRating rating={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('NumRating without rating renders correctly', () => {
    const tree = create(<NumRating />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
