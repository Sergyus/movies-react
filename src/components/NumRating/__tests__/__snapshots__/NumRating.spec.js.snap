// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumRating component renders correctly NumRating renders correctly 1`] = `
<span
  className="numRating"
>
  5
</span>
`;

exports[`NumRating component renders correctly NumRating without rating renders correctly 1`] = `null`;
